### ISSUE TITLE
[consistent-function-scoping] produces false-positive when there is no outer scope

### ISSUE THREAD

**Issue by nbouvrette** [sindresorhus/eslint-plugin-unicorn#2052]
The following code triggers an `Move arrow function 'toParametersString' to the outer scope.` [error](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v45.0.2/docs/rules/consistent-function-scoping.md) when there is no outer scope available.

```ts
    const toParametersString = (parameters: Record<string, any>): string =>
      Object.entries(parameters).reduce(
        (accumulator, [key, value]) =>
          typeof value === 'string' ? `${accumulator}&${key}=${value}` : accumulator,
        ''
      )
```

**Comment by fregante** [comment]
> when there is no outer scope available.

The indentation on that code snippet seems to suggest otherwise.  Closing for now.

Does it appear inside another function?
